src/vis/infowindow-manager.js

```diff
@@ -0,0 +1,103 @@
+var _ = require('underscore');
+var Overlay = require('./vis/overlay');
+var InfowindowModel = require('../geo/ui/infowindow-model');
+
+/**
+ * Manages the infowindows for a map. It listens to changes on the collection
+ * of layers and binds a new infowindow view/model to CartoDB.js whenever the
+ * collection of layers changes
+ */
+var InfowindowManager = function (vis) {
+  this._vis = vis;
+};
+
+InfowindowManager.prototype.manage = function (mapView, map) {
+  this._mapView = mapView;
+  this._map = map;
+
+  this._map.layers.bind('reset', function (layers) {
+    layers.each(this._addInfowindowForLayer, this);
+  }, this);
+  this._map.layers.bind('add', this._addInfowindowForLayer, this);
```

Comment:
Are you refering to [this](https://github.com/CartoDB/cartodb.js/pull/1132/files#diff-7d5ed7ef009eb2aac7940c8a0a158332R48)?